components/org.wso2.carbon.identity.organization.management.role.management.service/src/main/java/org/wso2/carbon/identity/organization/management/role/management/service/dao/RoleManagementDAOImpl.java

```diff
@@ -1109,10 +1109,17 @@ private void assignRoleAttributes(List<String> valueList, String roleId, String
             namedJdbcTemplate.withTransaction(template -> {
                 template.executeInsert(query,
                         namedPreparedStatement -> {
-                            for (int i = 0; i < numberOfGroups; i++) {
-                                namedPreparedStatement.setString(finalColumnName + i, valueList.get(i));
-                                namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_UM_ROLE_ID + i,
-                                        roleId);
+                            if (DB_SCHEMA_COLUMN_NAME_UM_PERMISSION_ID.equals(finalColumnName)) {
+                                for (int i = 0; i < numberOfGroups; i++) {
```

Comment:
```suggestion
                                for (int i = 0; i < valueList.size(); i++) {
```

Shall we make the variable more generic as it seems assigning set of groups but it could be either users, permissions also. Otherwise can use the suggested code. wdyt?

Reply:
resolved by 3f77b6501eea19535b19fa170e04087a557c17cd